server/config/strategies/facebook-auth.js

```diff
@@ -1,17 +1,48 @@
-/*var FacebookStrategy = require('passport-facebook').Strategy;
+var configAuth = require('./auth.js');
+var passport = require('passport');
+var User = require('mongoose').model('User');
+var FacebookStrategy = require('passport-facebook').Strategy;
+var LocalStrategy    = require('passport-local').Strategy;
 
-var FACEBOOK_APP_ID = 'your app id';  // Remove this from git repo
-var FACEBOOK_APP_SECRET = 'your app secret';  // Remove this from git repo
+module.exports = function() {
+  passport.use(new FacebookStrategy({
+          clientID: configAuth.facebookAuth.clientID,
+          clientSecret: configAuth.facebookAuth.clientSecret,
+          callbackURL: configAuth.facebookAuth.callbackURL,  //'http://greentourism.herokuapp.com/auth/facebook/callback',
+          profileFields: ['emails', 'photos', 'displayName', 'name', 'gender'],
+          passReqToCallback : false,
+          enableProof: true,
+          session: true,
+      },
+      function(accessToken, refreshToken, profile, done) {
+          process.nextTick(function() {
+              User.findOne({'providerData.facebook.id': profile.id}, function (err, user, eq, res, next) {
+                if (err) 
+                  return done(err);
+                if (user) {
+                    done(null, user);
+                } else {
+                  console.log(user)
+                  var user = new User();
+                  user.providerData.facebook.id = profile.id;
+                  user.providerData.facebook.token = accessToken;
+                  user.email = profile.emails[0].value;
+                  user.avatar = profile.photos[0].value;
+                  user.firstName = profile.name.givenName;
+                  user.lastName = profile.name.familyName;
 
-passport.use(new FacebookStrategy({
-    clientID: FACEBOOK_APP_ID,
-    clientSecret: FACEBOOK_APP_SECRET,
-    callbackURL: 'http://localhost:' + PORT + '/auth/facebook/callback'
-  },
-  function(accessToken, refreshToken, profile, done) {
-    User.findOrCreate(..., function(err, user) {
-      if (err) { return done(err); }
-      done(null, user);
-    });
+                  user.save(function(err) {
+                    if(err) {
+                      console.log(err);  //handle errors!
```

Comment:
Remove log.

User should have some feedback if there is an error, `console.log` doesn't give any of it.